gloo/test/memory_test.cc

```diff
@@ -11,18 +11,33 @@
 #include <fstream>
 #include <sstream>
 
+#ifdef _WIN32
+#include <Windows.h>
+#include <process.h>
+#endif
+
 namespace gloo {
 namespace test {
 namespace {
 
 size_t readResidentSetSize() {
   std::stringstream path;
+#ifdef _WIN32
+  path << "/proc/" << _getpid() << "/statm";
+#else
   path << "/proc/" << getpid() << "/statm";
+#endif
   std::ifstream f(path.str());
   size_t size;
   size_t resident;
   f >> size >> resident;
+#ifdef _WIN32
+  SYSTEM_INFO si;
+  GetSystemInfo(&si);
+  return (si.dwPageSize * resident);
+#else
   return (getpagesize() * resident);
+#endif
```

Comment:
Don't need `<process.h>`, and the current code won't work for getting the windows resident set size. On windows you can replace the entire function with:
```
PROCESS_MEMORY_COUNTERS counters{};
GetProcessMemoryInfo(GetCurrentProcess(), &counters, sizeof(counters));
return counters.WorkingSetSize;
```